package-build.el

```diff
@@ -325,19 +326,28 @@ A number as third arg means request confirmation if NEWNAME already exists."
 
 (defun package-build--checkout-darcs (name config dir)
   "Check package NAME with config CONFIG out of darcs into DIR."
-  (unless package-build-stable
-    (let ((repo (plist-get config :url)))
-      (with-current-buffer (get-buffer-create "*package-build-checkout*")
-        (cond
-         ((and (file-exists-p (expand-file-name "_darcs" dir))
-               (string-equal (package-build--darcs-repo dir) repo))
-          (package-build--princ-exists dir)
-          (package-build--run-process dir "darcs" "pull"))
-         (t
-          (when (file-exists-p dir)
-            (delete-directory dir t))
-          (package-build--princ-checkout repo dir)
-          (package-build--run-process nil "darcs" "get" repo dir)))
+  (let ((repo (plist-get config :url)))
+    (with-current-buffer (get-buffer-create "*package-build-checkout*")
+      (cond
+       ((and (file-exists-p (expand-file-name "_darcs" dir))
+             (string-equal (package-build--darcs-repo dir) repo))
+        (package-build--princ-exists dir)
+        (package-build--run-process dir "darcs" "pull" "--all"))
```

Comment:
Maybe the behavior of darcs has changed, because on the version I tested with (2.10.2), `pull` is interactive by default.